### PR TITLE
Neutron-damaged tungsten validation case from Dark et. al

### LINKED ISSUE #20: Neutron damaged tungsten validation
@jhdark would you want your recent paper on neutron damage to be in this V&V book?

### PR DESCRIPTION
**Pull request by rekomodo**
Resolves #20 . You might want to change the title for the case, it's not my best work.
I might be able to display the parameter tables a bit prettier, e.g. using latex, but I think they're fine for the time being.
Also, I can add the fitting procedure too, the code is available and we have no demonstration of FESTIM's fitting capabilities. I spent a long time toying with the code btw.

### CODE REVIEW

**Comment by rekomodo** [comment]
Weird how it's showing those previous commits, I must've messed something up on my side, I can fix it though. The first actual commit is https://github.com/festim-dev/V-V-report/pull/42/commits/5eb1f5255296d334e7a5fa3ff8b395715b029942

**Comment by rekomodo** [comment]
> Weird how it's showing those previous commits, I must've messed something up on my side, I can fix it though. The first actual commit is [5eb1f52](https://github.com/festim-dev/V-V-report/commit/5eb1f5255296d334e7a5fa3ff8b395715b029942)

Rebased the commit, looks good now.

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.16.2
+kernelspec:
+  display_name: vv-festim-report-env
+  language: python
+  name: python3
+---
+
+# Deuterium retention in neutron-damaged tungsten
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:15]:
```suggestion
# Deuterium retention in self-damaged tungsten
```

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+```{tags} 1D, TDS, trapping, transient
+```
+
++++
+
+This validation case is a thermo-desorption spectrum measurement perfomed by {cite}`dark_modelling_2024`.
+
+Several 0.8 mm thick samples of tungsten were damaged via annealing before being used to perform a TDS measurement.
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:24]:
```suggestion
Several 0.8 mm thick samples of tungsten were self-damaged and annealing before being used to perform a TDS measurement.
```

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+
+An ion beam with an incident flux of $5.79 \times 10^{19} \ \mathrm{D \ m^{-2} \ s^{-1}}$ was turned on for $72\mathrm{h}$ with an implantation temperature of $370\mathrm{K}$. The sample then rested for $12\mathrm{h}$ at $295\mathrm{K}$ before beggining the TDS measurement at $300\mathrm{K}$ with a temperature ramp of $0.05\mathrm{K}/s$.
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:26]:
```suggestion
An ion beam with an incident flux of $5.79 \times 10^{19} \ \mathrm{D \ m^{-2} \ s^{-1}}$ was turned on for $72 \ \mathrm{h}$ with an implantation temperature of $370 \ \mathrm{K}$. The sample then rested for $12 \ \mathrm{h}$ at $295 \ \mathrm{K}$ before beginning the TDS measurement at $300 \ \mathrm{K}$ with a temperature ramp of $0.05 \ \mathrm{K}/s$.
```

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+
+To reproduce this experiment, six traps are needed: 1 intrinsic trap and 5 neutron induced traps.
+The trap densities for the neutron induced traps were fitted by {cite}`dark_modelling_2024` for each _dpa_ amount using FESTIM's `NeutronInducedTrap`.
+
+The damage distribution for the neutron-induced traps is as follows:
+$$
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:32]:
```suggestion

$$
```

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+This validation case is a thermo-desorption spectrum measurement perfomed by {cite}`dark_modelling_2024`.
+
+Several 0.8 mm thick samples of tungsten were damaged via annealing before being used to perform a TDS measurement.
+
+An ion beam with an incident flux of $5.79 \times 10^{19} \ \mathrm{D \ m^{-2} \ s^{-1}}$ was turned on for $72\mathrm{h}$ with an implantation temperature of $370\mathrm{K}$. The sample then rested for $12\mathrm{h}$ at $295\mathrm{K}$ before beggining the TDS measurement at $300\mathrm{K}$ with a temperature ramp of $0.05\mathrm{K}/s$.
+
+To reproduce this experiment, six traps are needed: 1 intrinsic trap and 5 neutron induced traps.
+The trap densities for the neutron induced traps were fitted by {cite}`dark_modelling_2024` for each _dpa_ amount using FESTIM's `NeutronInducedTrap`.
+
+The damage distribution for the neutron-induced traps is as follows:
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:31]:
```suggestion
The damage distribution for the damage-induced traps is as follows:
```

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
++++
+
+This validation case is a thermo-desorption spectrum measurement perfomed by {cite}`dark_modelling_2024`.
+
+Several 0.8 mm thick samples of tungsten were damaged via annealing before being used to perform a TDS measurement.
+
+An ion beam with an incident flux of $5.79 \times 10^{19} \ \mathrm{D \ m^{-2} \ s^{-1}}$ was turned on for $72\mathrm{h}$ with an implantation temperature of $370\mathrm{K}$. The sample then rested for $12\mathrm{h}$ at $295\mathrm{K}$ before beggining the TDS measurement at $300\mathrm{K}$ with a temperature ramp of $0.05\mathrm{K}/s$.
+
+To reproduce this experiment, six traps are needed: 1 intrinsic trap and 5 neutron induced traps.
+The trap densities for the neutron induced traps were fitted by {cite}`dark_modelling_2024` for each _dpa_ amount using FESTIM's `NeutronInducedTrap`.
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:29]:
```suggestion
The trap densities for the neutron induced traps were fitted by {cite}`dark_modelling_2024` for each _dpa_ dose using FESTIM's `NeutronInducedTrap`.
```

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+import festim as F
+import numpy as np
+import sympy as sp
+import matplotlib.pyplot as plt
+
+# Parameters
+
+
+# Deuterium in Tungsten [35]
+D_0 = 1.6 * 1e-7 # m^2 s^-1
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:55]:
```suggestion
D_0 = 1.6e-7 # m^2 s^-1
```

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+# Setup
+import festim as F
+import numpy as np
+import sympy as sp
+import matplotlib.pyplot as plt
+
+# Parameters
+
+
+# Deuterium in Tungsten [35]
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:54]: I would avoid hard coding references like this since this index is likely to change. If you want to reference the paper for this diffusivity, please do it in a text block

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+D_0 = 1.6 * 1e-7 # m^2 s^-1
+E_D = 0.28 # eV
+w_atom_density = 6.3222e28
+
+sample_depth = 0.8e-3 # m
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:59]: I would use `sample_thickness`

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+plt.ylim(bottom=0, top=1e17)
+plt.ylabel(r"Desorption flux (m$^{-2}$ s$^{-1}$)")
+plt.xlabel(r"Temperature (K)")
+
+plt.show()
+```
+
+```{note}
+The experimental data was taken from {cite}`dark_modelling_2024` using [WebPlotDigitizer](https://automeris.io/).
+```
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:234]: You can use the actual data from [this folder](https://github.com/J-Dark-PhD/Modelling_neutron_damage_effects_in_tungsten/tree/v1.0/section_3_model_parameter_evaluation/data/tds_data_schwartz_selinger)

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 11]
```diff
@@ -0,0 +1,256 @@
...
+
+for i, trap in enumerate(model.traps):
+    for key, value in trap.__dict__.items():
+        glue(f'trap{i}{key}', value, display=False)
+    glue(f'ni{i}', n_i[i], display=False)
+```
+
+The density distribution of the neutron-induced traps is $n_i f(x)$.
+
+|Trap|k_0|E_k|E_p|p_0|n_i|
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:249]: Units are needed.
Also, does this table support mathtex? Eg. $k_0$

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 12]
```diff
@@ -0,0 +1,256 @@
...
+```
+
+```{code-cell} ipython3
+:tags: [hide-cell]
+
+from myst_nb import glue
+
+for i, trap in enumerate(model.traps):
+    for key, value in trap.__dict__.items():
+        glue(f'trap{i}{key}', value, display=False)
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:243]: I would try to format the strings here.

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 13]
```diff
@@ -0,0 +1,256 @@
...
+
+traps = [trap_1] + neutron_induced_traps
+model.traps = traps
+
+model.dt = F.Stepsize(
+    initial_value=1,
+    stepsize_change_ratio=1.1,
+    t_stop=t_imp + t_rest * 0.5,
+    dt_min=1e-1,
+    stepsize_stop_max=50,
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:146]: This triggers a deprecation warning:

```
/home/docs/checkouts/readthedocs.org/user_builds/festim-vv-report/conda/42/lib/python3.11/site-packages/festim/stepsize.py:54: DeprecationWarning: stepsize_stop_max and t_stop attributes will be deprecated in a future release, please use max_stepsize instead
  warnings.warn(
```

Instead please use the new `max_stepsize` parameter (see [here](https://festim.readthedocs.io/en/stable/userguide/stepsize.html#stepsize) for more details)

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 14]
```diff
@@ -0,0 +1,256 @@
...
+)
+
+model.exports = [derived_quantities]
+
+model.initialise()
+model.run()
+```
+
+```{note}
+Implementation details from {cite}`dark_modelling_2024_code`.
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:178]: In reality the implementation details are available in the paper, this will be a copy of the initial implementation. Since you will add the other doses, I would rephrase this as "Experimental data taken from [...]"

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 15]
```diff
@@ -0,0 +1,256 @@
...
+
+from myst_nb import glue
+
+for i, trap in enumerate(model.traps):
+    for key, value in trap.__dict__.items():
+        glue(f'trap{i}{key}', value, display=False)
+    glue(f'ni{i}', n_i[i], display=False)
+```
+
+The density distribution of the neutron-induced traps is $n_i f(x)$.
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:247]:
```suggestion
The density distribution of the neutron-induced traps is $n_i \ f(x)$.
```

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 16]
```diff
@@ -0,0 +1,256 @@
...
+temp = min_temp + Beta * (t - start_tds)
+
+# plotting simulation data
+plt.plot(temp, flux_total, linewidth=3, label="FESTIM")
+
+# plotting trap contributions
+trap_data = [derived_quantities.filter(fields=f"{i}").data for i in range(1, 7)]
+contributions = [-np.diff(trap) / np.diff(t) for trap in trap_data]
+
+colors = [(0.9*(i % 2), 0.2*(i % 4), 0.4*(i % 3)) for i in range(6)]
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:205]: Just sharing an awesome tool made by @holtzy: [PyPalettes](https://python-graph-gallery.com/color-palette-finder/) which helps finding cool colour palettes!

I quite like [this one](https://python-graph-gallery.com/color-palette-finder/?palette=acadia) for instance.

The python code for this is:

```python
from pypalettes import load_cmap
cmap = load_cmap("Acadia")
```

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 17]
```diff
@@ -0,0 +1,256 @@
...
+---
+
+# Deuterium retention in neutron-damaged tungsten
+
+```{tags} 1D, TDS, trapping, transient
+```
+
++++
+
+This validation case is a thermo-desorption spectrum measurement perfomed by {cite}`dark_modelling_2024`.
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:22]:
```suggestion
This validation case is a thermo-desorption spectrum measurement on damaged tungsten. The complete description is available in {cite}`dark_modelling_2024`.
```

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 18]
```diff
@@ -0,0 +1,256 @@
...
+
+
+# Deuterium in Tungsten [35]
+D_0 = 1.6 * 1e-7 # m^2 s^-1
+E_D = 0.28 # eV
+w_atom_density = 6.3222e28
+
+sample_depth = 0.8e-3 # m
+
+# Table 2
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:61]:
```suggestion
# Table 2 from Dark et al 10.1088/1741-4326/ad56a0
```

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 19]
```diff
@@ -0,0 +1,256 @@
...
+sigma = 0.5e-9 # m
+t_imp = 72 * 3600 # s
+t_rest = 12 * 3600 # s
+Beta = 0.05 # K s^-1
+
+fluence = 1.5e25
+flux = fluence / t_imp
+
+# Deuterium Beam Profile (S = flux * f(x))
+distribution = (1 / (sigma * (2 * np.pi) ** 0.5) * sp.exp(-0.5 * ((F.x - R_p) / sigma) ** 2))
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:74]: I would maybe define this right before the source

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 20]
```diff
@@ -0,0 +1,256 @@
...
+        k_0 = k_0,
+        E_k = E_D,
+        p_0 = 1e13,
+        E_p = E_p[i],
+        density = n_i[i] * damage_dist,
+        materials = damaged_tungsten,
+    ))
+
+traps = [trap_1] + neutron_induced_traps
+model.traps = traps
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:139]: Can you condense this to

```suggestion
model.traps = [trap_1] + neutron_induced_traps
```

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
Hi @rekomodo very great job!

Here's a first round of comments. I'll also invite @jhdark to pitch in since it's his work

- Can you format the python code with black?
- Can you add the other dpa doses (including 0 dpa)?

File: report/validation/thermodesorption_spectra/dark/dark_tds.md (hunk 16 above)
Comment (rekomodo) [reply report/validation/thermodesorption_spectra/dark/dark_tds.md:205]: That is very cool, thanks!

File: report/validation/thermodesorption_spectra/dark/dark_tds.md (hunk 11 above)
Comment (rekomodo) [reply report/validation/thermodesorption_spectra/dark/dark_tds.md:252]: It does. I added the units using it as well.

**Comment by rekomodo** [comment]
The FESTIM data is off for the 2.5 dpa case in the second figure which replicates figure 4a) from the paper. The other dpa's seem fine but I did not check thoroughly. I'll look into it, but I'd appreciate if you had a look first.

**Comment by RemDelaporteMathurin** [comment]
> The FESTIM data is off for the 2.5 dpa case 

Have you checked all the parameters compared to the scripts given with @jhdark paper?

**Comment by rekomodo** [comment]
> > The FESTIM data is off for the 2.5 dpa case
> 
> Have you checked all the parameters compared to the scripts given with @jhdark paper?

Yeah I've checked multiple times, thought it might be an implementation thing. I'll check thoroughly again soon.

**Comment by RemDelaporteMathurin** [comment]
You must have changed something because I ran the script in the repository directly and obtained the correct spectra:

image

**Comment by RemDelaporteMathurin** [comment]
@rekomodo everything is fine. On your end at least. After a few hours of investigating, @jhdark and I realised there was an issue in the paper repository. All the dpa levels were exported to one folder
https://github.com/J-Dark-PhD/Modelling_neutron_damage_effects_in_tungsten/blob/95d5cbff34364d63bb1e3dd7492fa492578a452a/section_3_model_parameter_evaluation/generate_data.py#L193

except for the 2.5 dpa where it was accidentally exported to another folder
https://github.com/J-Dark-PhD/Modelling_neutron_damage_effects_in_tungsten/blob/95d5cbff34364d63bb1e3dd7492fa492578a452a/section_3_model_parameter_evaluation/generate_data.py#L204

The 2.5 dpa case that was plotted was in fact old data that had been pushed to the repo at some point and never updated because festim was writing elsewhere.

This is the reason why one should not push generated files to a repository, and why this executable book is the way to go 🚀

**Comment by RemDelaporteMathurin** [comment]
Could you try tweaking the densities of the first two traps to better fit the data?

File: report/validation/thermodesorption_spectra/dark/dark-original.csv
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark-original.csv:1]: is this used anywhere?

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 21]
```diff
@@ -0,0 +1,372 @@
...
+```
+
+```{code-cell} ipython3
+:tags: [hide-input]
+
+# Color Bar
+from matplotlib import cm, colors
+
+#norm = colors.LogNorm(vmin=min(dpa_values[1:]), vmax=max(dpa_values))
+norm = colors.LogNorm(1, 2)
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:306]: This makes the curve all look blackish. I would reverse the norm

File: report/validation/thermodesorption_spectra/dark/dark_tds.md
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:1]: There is too much code duplication here. You only need to run each dpa case once, but make an additional plot for one of them. 

You need several functions: 
`run_tds(densities)` : runs a festim model for a given set of **damage-induced** trap densities and returns the `DerivedQuantities` object
`plot_simulated_tds(derived_quantities, trap_contributions: bool, **kwargs)`: plots a simulated TDS from a `DerivedQuantities` object, `kwargs` are the keyword arguments for `plt.plot(x, y, **kwargs)`. If `trap_contributions` is True then also plot the grey areas for each trap.
`plot_experimental_tds(dpa, **kwargs)` : plots the experimental tds based on a dpa level
`scatter_experimental_tds(dpa, **kwargs)`: same as `plot_experimental_tds` but for a scatter plot.


Once you have all these functions, the implementation is simply:

```python
dpa_to_densities = {
    0.001: [1e+24, 2.5e+24, 1e+24, 1e+24, 2e+23], 
    0.005: [3.5e+24, 5e+24, 2.5e+24, 1.9e+24, 1.6e+24], 
    0.023: [2.2e+25, 1.5e+25, 6.5e+24, 2.1e+25, 6e+24], 
    0.1: [4.8e+25, 3.8e+25, 2.6e+25, 3.6e+25, 1.1e+25], 
    0.23: [5.4e+25, 4.4e+25, 3.6e+25, 3.9e+25, 1.4e+25], 
    0.5: [5.5e+25, 4.6e+25, 4e+25, 4.5e+25, 1.7e+25], 
    2.5: [5.8e+25, 6.5e+25, 4.5e+25, 5.5e+25, 2e+25], # re-fit
}

for dpa, densities in dpa_to_densities.items():
    derived_quantities = run_tds(densities)
    plt.figure(1)
    plot_experimental_tds(dpa)
    plot_simulated_tds(derived_quantities)
    if dpa == 0.02:
        plt.figure(2)
        plot_simulated_tds(derived_quantities, trap_contribution=True)
        scatter_experimental_tds(dpa)
```

File: report/validation/thermodesorption_spectra/dark/dark_tds.md [hunk 22]
```diff
@@ -0,0 +1,372 @@
...
+
+### Density per induced trap for each dpa dose
+
+|Trap|$0.001$|$0.005$|$0.023$|$0.1$|$0.23$|$0.5$|$2.5$|
+|:---|:------|:------|:------|:----|:-----|:----|:----|
+|D1|{glue:text}`ni_0_0:.2e`|{glue:text}`ni_0_1:.2e`|{glue:text}`ni_0_2:.2e`|{glue:text}`ni_0_3:.2e`|{glue:text}`ni_0_4:.2e`|{glue:text}`ni_0_5:.2e`|{glue:text}`ni_0_6:.2e`|
+|D2|{glue:text}`ni_1_0:.2e`|{glue:text}`ni_1_1:.2e`|{glue:text}`ni_1_2:.2e`|{glue:text}`ni_1_3:.2e`|{glue:text}`ni_1_4:.2e`|{glue:text}`ni_1_5:.2e`|{glue:text}`ni_1_6:.2e`|
+|D3|{glue:text}`ni_2_0:.2e`|{glue:text}`ni_2_1:.2e`|{glue:text}`ni_2_2:.2e`|{glue:text}`ni_2_3:.2e`|{glue:text}`ni_2_4:.2e`|{glue:text}`ni_2_5:.2e`|{glue:text}`ni_2_6:.2e`|
+|D4|{glue:text}`ni_3_0:.2e`|{glue:text}`ni_3_1:.2e`|{glue:text}`ni_3_2:.2e`|{glue:text}`ni_3_3:.2e`|{glue:text}`ni_3_4:.2e`|{glue:text}`ni_3_5:.2e`|{glue:text}`ni_3_6:.2e`|
+|D5|{glue:text}`ni_4_0:.2e`|{glue:text}`ni_4_1:.2e`|{glue:text}`ni_4_2:.2e`|{glue:text}`ni_4_3:.2e`|{glue:text}`ni_4_4:.2e`|{glue:text}`ni_4_5:.2e`|{glue:text}`ni_4_6:.2e`|
```
Comment (RemDelaporteMathurin) [inline report/validation/thermodesorption_spectra/dark/dark_tds.md:372]: We need to modify this as it's a bit redundant.

Just two tables:

1) for each damage induced trap, all the trap parameters except for density
2) for each damage level, for each trap, density

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
I think we need quite a bit of refactoring. I'll branch out of this and open a PR to your branch with changes, is that ok?

File: report/validation/thermodesorption_spectra/dark/dark-original.csv
Comment (rekomodo) [reply report/validation/thermodesorption_spectra/dark/dark-original.csv:1]: no, 0 purpose. I forgot to remove it

File: report/validation/thermodesorption_spectra/dark/dark_tds.md (hunk 22 above)
Comment (rekomodo) [reply report/validation/thermodesorption_spectra/dark/dark_tds.md]: Do you want me to re-add the tables with this configuration or leave them out? I could also do the data frames experiment now.

Comment (RemDelaporteMathurin) [reply report/validation/thermodesorption_spectra/dark/dark_tds.md]: As you prefer :-)

Comment (rekomodo) [reply report/validation/thermodesorption_spectra/dark/dark_tds.md]: I'm trying out the pandas tables, sadly I can't figure out how to display some of their text as latex. They can be converted to markdown, though, so that's a way if we really need it.

Comment (RemDelaporteMathurin) [reply report/validation/thermodesorption_spectra/dark/dark_tds.md]: This seems related to https://github.com/executablebooks/jupyter-book/issues/1501

**Review by RemDelaporteMathurin — APPROVED** [review]
@rekomodo I just added an annotation for the undamaged case in the second graph.

Otherwise looks good to me!
